src/libsystemd-network/dhcp-network.c

```diff
@@ -113,9 +113,10 @@ static int _bind_raw_socket(int ifindex, union sockaddr_union *link,
         link->ll.sll_ifindex = ifindex;
         link->ll.sll_hatype = htobe16(arp_type);
         link->ll.sll_halen = mac_addr_len;
+        memset(link->ll.sll_addr, 0, sizeof(link->ll.sll_addr));
```

Comment:
instead of implicitly invoking memset() here, I think it would be much nicer to rewrite the six lines above to become:

```c
link->ll = (struct sockaddr_ll) {
        .sll_family = AF_PACKET,
        .sll_protocol = htobe16(ETH_P_IP),
        .sll_ifindex = ifindex,
        .sll_hatype = htobe16(arp_type),
        .sll_halen = mac_addr_len,
};
```

This is not only more readable I think, but also has the nice effect of zero-initializing everything not mentioned explicitly.